### PR TITLE
Add Prometheus Exporter: Step 0 - Update MetricsExporter Interface

### PR DESCRIPTION
**Pull request by erichsueh3**
This PR implements headers in the [sdk/metrics](https://github.com/open-telemetry/opentelemetry-cpp/tree/master/sdk/include/opentelemetry/sdk/metrics) folder that lays out the groundwork for the Prometheus Exporter project. The Metrics interface, exporter.h, is the interface which the PrometheusExporter class is derived from. The ReturnCode class, return_code.h, lays out the possible return values returned when calling Export() from the PrometheusExporter class. 

This PR is the first in a series of PRs that aims to add a full implementation of a Prometheus Exporter within the OpenTelemetry C++ Repository.

Note to reviewers: Please keep in mind that there are other PRs (yet to be filed) related to this one so codecov tests may not fully pass.

### CODE REVIEW

**Comment by erichsueh3** [comment]
To align our Metrics Exporter Interface with the previously existing Metrics Exporter Interface, we have removed record_code.h and added those codes to ExportResult in the previously existing exporter.h. Additionally, since there is no Shutdown function in the Metrics specification, we have removed that from our interface and aligned it with the existing interface.

**Comment by reyang** [comment]
@erichsueh3 please rebase and resolve the conflicts. Thanks.

File: sdk/include/opentelemetry/sdk/metrics/exporter.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+   * concurrently for the same exporter instance.
+   * @param records a vector of unique pointers to metric records
+   */
+  virtual ExportResult Export(const std::vector<Record> &records) noexcept = 0;
+
+  /**
+   * In the Metrics specification, there is no Shutdown function required for exporters
+   * The Shutdown function can be implemented within exporters that wish to have one,
+   * but will not be enforced through this header
+   */
```
Comment (pyohannes) [inline sdk/include/opentelemetry/sdk/metrics/exporter.h:73]: This doesn't make sense and we should remove it.

The `MetricsExporter` is passed on to the `Controller` as a `std::unique_ptr` (see [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/include/opentelemetry/sdk/metrics/controller.h#L30)). With that design, the user doesn't have the possibility to call any `Shutdown` method on the `MetricsExporter`, only the `Controller` could do so (but cannot if the interface doesn't have a `Shutdown` method).

I recommend removing this comment for now. We can extend the exporter interface by an optional shutdown function later on.

Comment (erichsueh3) [reply sdk/include/opentelemetry/sdk/metrics/exporter.h:73]: This comment is originally included within the current `exporter.h` file, but I agree; will remove this comment for now.

**Comment by erichsueh3** [comment]
> @erichsueh3 please rebase and resolve the conflicts. Thanks.

@reyang rebased and resolved the conflicts, looks like all the checks are passing as well.

**Pull request merged**